plugins/connection/aws_ssm.py

```diff
@@ -343,7 +360,11 @@ def start_session(self):
         ssm_parameters = dict()
         client = self._get_boto_client('ssm', region_name=region_name, profile_name=profile_name)
         self._client = client
-        response = client.start_session(Target=self.instance_id, Parameters=ssm_parameters)
+        start_session_args = dict(Target=self.instance_id, Parameters=ssm_parameters)
+        document_name = self.get_option('ssm_document')
+        if document_name is not None:
+            start_session_args['DocumentName'] = document_name
+        response = client.start_session(**start_session_args)
```

Comment:
Could you please add an integration test for this feature idea?

Reply:
Sure, I tried something, let's see if it's working. Ideally, we'd create a custom SSM document, but unfortunately I can't find an ansible module for that, so instead, I just put the default SSM document used behind the scene by the SSM client